lib/iris/experimental/regrid.py

```diff
@@ -491,6 +491,31 @@ def _regrid_area_weighted_array(
         cached_x_bounds.append(x_bounds)
         cached_x_indices.append(x_indices)
 
+    # Move y_dim and x_dim to last dimensions
+    x_dim_orig = copy.copy(x_dim)
+    y_dim_orig = copy.copy(y_dim)
+    if x_dim is None and y_dim is None:
+        # e.g. a scalar point such as a vertical profile
+        pass
+    elif x_dim is not None and y_dim is None:
+        # test cross_section along line longitude
+        src_data = np.moveaxis(src_data, x_dim, -1)
+        x_dim = src_data.ndim - 1
+    elif y_dim is not None and x_dim is None:
+        # test cross_section along line latitude
+        src_data = np.moveaxis(src_data, y_dim, -1)
+        y_dim = src_data.ndim - 1
+    elif x_dim < y_dim:
+        src_data = np.moveaxis(src_data, x_dim, -1)
+        src_data = np.moveaxis(src_data, y_dim - 1, -2)
```

Comment:
Shouldn't `y_dim - 1` be just `y_dim`? In fact, can the last two loops be combined? Regardless of whether `x_dim` is greater or less than `y_dim`, the result is the same?

Reply:
Yes, as y gets shifted along one position as x moves to the end.